### PR TITLE
Multiple Tags Cleanup

### PR DESCRIPTION
**Pull request by TWilson023**
Cleaning up old `tagId` while maintaining backwards compatibility

### Test Plan
Ensure the following flows work and endpoints return `tags` *and* `tagId` as expected:
- [x] Creating link with tags
- [x] Creating link without tags
- [x] Creating link with `tagNames`
- [x] Editing link with tags
- [x] Editing link without tags
- [x] Editing link with `tagNames`
- [x] Bulk creating links with tags
- [x] Bulk creating links without tags
- [x] Bulk creating links with `tagNames`

### CODE REVIEW

**Comment by devkiran** [comment]
@TWilson023 Could you please review the conflicts and resolve them?

**Review by devkiran — APPROVED** [review]

File: apps/web/app/api/links/[linkId]/route.ts [hunk 1]
```diff
@@ -52,8 +52,14 @@ export const PUT = withAuth(async ({ req, headers, workspace, link }) => {
...
   const body = updateLinkBodySchema.parse(await req.json());
 
+  // Add body onto existing link but maintain NewLinkProps form for processLink
   const updatedLink = {
     ...link,
+    expiresAt:
+      link.expiresAt instanceof Date
+        ? link.expiresAt.toISOString()
+        : link.expiresAt,
+    geo: link.geo as NewLinkProps["geo"],
```
Comment (steven-tey) [inline apps/web/app/api/links/[linkId]/route.ts:62]: @TWilson023 we can't do this because users won't be able to update the `expiresAt` and `geo` params for their links

File: apps/web/lib/api/links/process-link.ts [hunk 2]
```diff
@@ -261,9 +302,7 @@ export async function processLink({
       // make sure projectId is set to the current workspace
       projectId: workspace?.id || null,
       // if userId is passed, set it (we don't change the userId if it's already set, e.g. when editing a link)
-      ...(userId && {
-        userId,
-      }),
+      userId: userId || payload.userId || null,
```
Comment (steven-tey) [inline apps/web/lib/api/links/process-link.ts]: why is this change required?

File: apps/web/lib/api/links/process-link.ts [hunk 3]
```diff
@@ -221,9 +260,11 @@ export async function processLink({
     };
   }
 
+  let expiresAt: Date | null = null;
+
   // expire date checks
-  if (expiresAt) {
-    const datetime = parseDateTime(expiresAt);
+  if (expiresAtRaw) {
+    const datetime = parseDateTime(expiresAtRaw);
```
Comment (steven-tey) [inline apps/web/lib/api/links/process-link.ts]: Do we still need this change?

**Review by steven-tey — CHANGES_REQUESTED** [review]

File: apps/web/app/api/links/[linkId]/route.ts (hunk 1 above)
Comment (TWilson023) [reply apps/web/app/api/links/[linkId]/route.ts:62]: Won't these be overridden by `...body` below if users update them?

File: apps/web/lib/api/links/process-link.ts (hunk 2 above)
Comment (TWilson023) [reply apps/web/lib/api/links/process-link.ts]: Actually I think we can just make `userId` optional on `ProcessedLinkProps` instead ([commit](https://github.com/dubinc/dub/pull/631/commits/e479f14bfa02be6adafc1e497613b15bb5d32dea))

File: apps/web/lib/api/links/process-link.ts (hunk 3 above)
Comment (TWilson023) [reply apps/web/lib/api/links/process-link.ts]: Just updated this to be less complicated ([commit](https://github.com/dubinc/dub/pull/631/commits/39b151a7470c64d9edbf0dfc9d0eca86ef0cba94))

File: apps/web/app/api/links/[linkId]/route.ts (hunk 1 above)
Comment (steven-tey) [reply apps/web/app/api/links/[linkId]/route.ts:62]: oh yeah good point!

**Review by steven-tey — APPROVED** [review]

**Review by criskell — APPROVED** [review]
